lib/DAV/Auth/Backend/Esn.php

```diff
@@ -113,64 +110,32 @@ private function decodeResponse($response) {
         return [true, $type];
     }
 
-
-    # <Modified by xguimard>
-    protected function validateUserPassLDAP($username, $ldapBase, $password) {
+    protected function validateUserPass($username, $password) {
         $user = trim($username);
 
-        # Open LDAP connection
-        $ldapCon = ldap_connect(LDAP_SERVER);
-        if (!$ldapCon) {
-            error_log('Unable to connect to LDAP server');
-            return [false];
-        }
-        ldap_set_option($ldapCon, LDAP_OPT_PROTOCOL_VERSION, 3);
-        ldap_set_option($ldapCon, LDAP_OPT_REFERRALS, 0);
-
-        # Try to authenticate
-        $ldapBind = ldap_bind($ldapCon,  "uid=$username," . $ldapBase, $password);
-
-        if (!$ldapBind) {
-            error_log("Bad credentials");
-            return [false];
-        }
+        $adminPrefix = SABRE_ADMIN_LOGIN . '&';
+        if (strlen(SABRE_ADMIN_LOGIN) > 0 && strpos($user, $adminPrefix) === 0) {
+            $requestedUsername = substr($user, strlen($adminPrefix));
+            if ($password != SABRE_ADMIN_PASSWORD) {
+                error_log('Bad admin password.');
+                return [false];
+            }
```

Comment:
So `$password` is valid only if == `SABRE_ADMIN_PASSWORD` ? No more ldap->bind ?

Reply:
No LDAP bind for the ADMIN, yes